Subject: Fix preempt-none build errors
From: Tony Jones <tonyj@suse.de>
Upstream: no

Fix build errors when building CONFIG_PREEMPT_NONE.

--- linux-2.6.27-RT.orig/drivers/base/class.c
+++ linux-2.6.27-RT/drivers/base/class.c
@@ -148,7 +148,11 @@ int __class_register(struct class *cls,
 	INIT_LIST_HEAD(&cp->class_devices);
 	INIT_LIST_HEAD(&cp->class_interfaces);
 	kset_init(&cp->class_dirs);
+#ifdef CONFIG_PREEMPT_RT
 	_mutex_init(&cp->class_mutex, "struct class mutex", key);
+#else
+	__mutex_init(&cp->class_mutex, "struct class mutex", key);
+#endif
 	error = kobject_set_name(&cp->class_subsys.kobj, "%s", cls->name);
 	if (error) {
 		kfree(cp);
--- linux-2.6.27-RT.orig/include/linux/rt_lock.h
+++ linux-2.6.27-RT/include/linux/rt_lock.h
@@ -138,11 +138,11 @@ typedef struct {
 
 typedef raw_rwlock_t rwlock_t;
 
-#define __RW_LOCK_UNLOCKED	_RAW_RW_LOCK_UNLOCKED
+#define __RW_LOCK_UNLOCKED	(rwlock_t)_RAW_RW_LOCK_UNLOCKED
 
 #endif
 
-#define RW_LOCK_UNLOCKED	__RW_LOCK_UNLOCKED(rw_old_style)
+#define RW_LOCK_UNLOCKED	(rwlock_t)__RW_LOCK_UNLOCKED(rw_old_style)
 
 
 #define DEFINE_RWLOCK(name) \
